Doppler.Sap/Factory/CreditNoteHandler.cs

```diff
@@ -76,12 +85,93 @@ public async Task<SapTaskResult> Handle(SapTask dequeuedTask)
             }
         }
 
+        public async Task<SapTaskResult> UpdatePaymentStatusHandle(SapTask dequeuedTask)
+        {
+            try
+            {
+                var sapSystem = SapSystemHelper.GetSapSystemByBillingSystem(dequeuedTask.CreditNoteRequest.BillingSystemId);
+                var sapTaskHandler = _sapServiceSettingsFactory.CreateHandler(sapSystem);
+                var existentCreditNote = await sapTaskHandler.TryGetCreditNoteByCreditNoteId(dequeuedTask.CreditNoteRequest.CreditNoteId);
+
+                if (existentCreditNote == null)
+                {
+                    return new SapTaskResult
+                    {
+                        IsSuccessful = false,
+                        SapResponseContent = $"Credit Note could'n update to SAP because the credit note does not exist: '{dequeuedTask.CreditNoteRequest.CreditNoteId}'.",
+                        TaskName = "Updating Credit Note Request"
+                    };
+                }
+
+                var sapResponse = await UpdateCreditNote(existentCreditNote, dequeuedTask, sapSystem);
+                if (!sapResponse.IsSuccessful)
+                {
+                    _logger.LogError($"Credit Note could'n update to SAP because exists an error: '{sapResponse.SapResponseContent}'.");
+                    return sapResponse;
+                }
+                else if (dequeuedTask.CreditNoteRequest.Type != (int)CreditNoteEnum.Refund || !dequeuedTask.CreditNoteRequest.TransactionApproved)
```

Comment:
This `else `sentence is unnecessary, you can remove it

Reply:
Done